third_party/amd/lib/TritonAMDGPUTransforms/Utility.h

```diff
@@ -16,4 +18,13 @@ using namespace mlir;
 int deduceMinCountOnDefChain(Value defValue, Operation *consumerOp,
                              llvm::function_ref<int(Operation *)> countFunc);
 
+// Returns a padded shared encoding minimizing bank conflicts for the given
+// tensor and dot encoding.
+std::optional<triton::gpu::PaddedSharedEncodingAttr>
```

Comment:
Typically we don't wrap attributes with `std::optional`--you can use null attribute to mean failures.